---
title: C6031
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6031
helpviewer_keywords:
- C6031
ms.assetid: 59e1ef0a-b3ca-4ffa-bcb3-ad2bd22ece22
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: e1e9345c2568a13d14e6781c51123097617b6466
ms.sourcegitcommit: 535ef05b1e553f0fc66082cd2e0998817eb2a56a
ms.translationtype: MT
ms.contentlocale: ja-JP
ms.lasthandoff: 10/07/2019
ms.locfileid: "72013889"
---
# <a name="c6031"></a>C6031
警告 C6031: 戻り値が無視されました: @no__t 0function > 予期しない値を返す可能性があります

 この警告は、呼び出し元の関数が、戻り値を使用してエラーを通知する関数呼び出しの戻り値をチェックしていないことを示します。 呼び出されている関数によっては、この不具合によって、クラッシュや、エラー状態のデータ破損やリソース不足の状況など、一見ランダムなプログラム misbehavior が発生する可能性があります。

 一般に、ディスク、ネットワーク、メモリなどのリソースを必要とする関数の呼び出しは常に成功すると想定するのは安全ではありません。 呼び出し元は常に戻り値を確認し、エラーケースを適切に処理する必要があります。 また、値が有用な方法で検査されることを確認する @no__t 0 の注釈を使用することも検討してください。

## <a name="example"></a>例
 この警告が発生するコード例を次に示します。

```cpp
#include <stdio.h>
void f( )
{
  fopen( "test.c", "r" ); // return value ignored
  // code ...
}
```

 この警告を解決するには、次のコードに示すように、関数の戻り値を確認します。

```cpp
#include <stdio.h>
void f( )
{
  FILE *stream;
  if((stream = fopen( "test.c", "r" )) == NULL )
    return;
  // code ...
}
```

 次のコードでは、セーフ関数 `fopen_s` を使用して、この警告を修正しています。

```cpp
#include <stdio.h>
void f( )
{
  FILE *stream;
  errno_t err;

  if( (err  = fopen_s( &stream, "test.c", "r" )) !=0 )
  {
    // code ...
  }
}
```

 また、次のコードに示すように、呼び出し元が `_Check_return_` プロパティで注釈が付けられた関数の戻り値を無視した場合にも、この警告が生成されます。

```cpp

#include <sal.h>
_Check_return_ bool func();

void test_f()
{
    func(); //  Warning C6031
}
```

 前の警告を修正するには、次のコードに示すように戻り値を確認します。

```cpp

#include <sal.h>
_Check_return_ bool func();

void test_f()
{
    if( func() ) {
        // code ...
    }
}
```

## <a name="see-also"></a>関連項目

- [fopen_s、_wfopen_s](/cpp/c-runtime-library/reference/fopen-s-wfopen-s)
- [SAL 注釈を使用したコード障害の軽減](using-sal-annotations-to-reduce-c-cpp-code-defects.md)
